src/bin/ion/commands/generate/model.rs

```diff
@@ -170,14 +170,13 @@ impl FullyQualifiedTypeReference {
 #[derive(Debug, Clone, PartialEq, Serialize)]
 pub enum AbstractDataType {
     // Represents a scalar type which also has a name attached to it and is nominally distinct from its base type.
-    #[allow(dead_code)]
     WrappedScalar(WrappedScalar),
     // Represents a scalar value (e.g. a string or integer or user defined type)
-    #[allow(dead_code)]
     Scalar(Scalar),
     // A series of zero or more values whose type is described by the nested `element_type`
-    #[allow(dead_code)]
     Sequence(Sequence),
+    // Represents a sequence type which also has anme attached to it and is nominally distinct from its element type.
+    WrappedSequence(WrappedSequence),
```

Comment:
Are we going to have to create corresponding `Wrapped*` variants for all of the non-wrapped variants? Why/why not?

Reply:
No. I think its in general there are only 3 types structure, sequence and scalar. Where only nested structure would have a name but scalar and sequence would not have name. Hence to separate out whether they are named or not I have them as a separate variant. I think having them separate allows us to perform specific operations for these types which will be stored as property instead of entirely different class in the generated code.

e.g. For below ISL:
```
type::{
   name: my_type,
   fields: {
        foo: {type: list, element: string} // this will not create a new nested class in the generated code
   }
}
```
Here's the generated code in Java:
```
class MyType {
   private java.util.ArrayList<String> foo;
   ...
}
```

whereas for structure type like following:
```
type::{
   name: my_type,
   fields: {
        foo: {
           fields: {
               bar: string
           }
        } 
   }
}
```
Here's the generated code in Java, which creates a new nested type for this nested structure:
```
class MyType {
   private NestedType1 foo;
   ...
   
   class NestedType1 {
       private String bar;
       ...
   }
}
```

Reply:
Do we really need the non-wrapped variant of sequence? We have parameterized, fully-qualified type references, so we should be able to model the non-wrapped lists as e.g. `java.util.List<org.example.MyGeneratedClass>`.

Reply:
`WrpapedSequence` use `FullyQualifiedTypeName` (not `FullyqualifiedTypeReference`). And `WrappedSequence` have a name, for example:
```
type::{
   name: foo, // this is the name of the wrapped sequence
   type: list,
   element: string
}
```
Above wrapped sequence's name will be stored as `org.example.Foo` .
Whereas `Sequence` would not have a name, for example:
```
{type: list, element: string} // does not have a name
```
We can surely store it as fully-qualified type references and thats what we already we store the `element` property as `FullyWualfiiedTypeReference` and then when storing this type into `TypeStore` we use `java.util.ArrayList<String>` but it still wouldn't have name (like the wrapped sequence has name as `foo`)

Reply:
Sure, but the non-wrapped version (`Sequence`) does not have a name and does not have a class generated for it. Therefore, why do we need to have the unnamed `Sequence` model? You don't need a reference to the `Sequence` when you can just use a parameterized reference such as `java.util.ArrayList<java.lang.String>`.

For example, if you have this schema, the `{ type: list, element: string }` does not need a `Sequence` to be created to represent it. Instead, the field `bar` should have the data model equivalent of a reference to `java.util.ArrayList<java.lang.String>`.
```
type::{
  name: foo,
  fields: {
    bar: { type: list, element: string }
  }
}
```

Reply:
I have created an issue for this: https://github.com/amazon-ion/ion-cli/issues/145. Need to look at if removing the non-wrapped variants has any requirements for how they are rendered in templates.